dist/src/main/java/io/zeebe/broker/StandaloneBroker.java

```diff
@@ -9,24 +9,52 @@
 
 import static java.lang.Runtime.getRuntime;
 
+import io.zeebe.EnvironmentHelper;
 import io.zeebe.broker.system.configuration.BrokerCfg;
+import io.zeebe.legacy.tomlconfig.LegacyConfigurationSupport;
+import io.zeebe.legacy.tomlconfig.LegacyConfigurationSupport.Scope;
 import io.zeebe.util.FileUtil;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.util.concurrent.CountDownLatch;
 import org.apache.logging.log4j.LogManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.core.env.Environment;
 
-public class StandaloneBroker {
-  private static final CountDownLatch WAITING_LATCH = new CountDownLatch(1);
-  private static String tempFolder;
+@SpringBootApplication
+public class StandaloneBroker implements CommandLineRunner {
+
+  @Autowired BrokerCfg configuration;
+  @Autowired Environment springEnvironment;
+
+  private final CountDownLatch waiting_latch = new CountDownLatch(1);
+  private String tempFolder;
 
   public static void main(final String[] args) throws Exception {
+    System.setProperty("spring.banner.location", "classpath:/assets/zeebe_broker_banner.txt");
+
+    final LegacyConfigurationSupport legacyConfigSupport =
+        new LegacyConfigurationSupport(Scope.BROKER);
+    legacyConfigSupport.checkForLegacyTomlConfigurationArgument(args, "broker.cfg.yaml");
+    legacyConfigSupport.checkForLegacyEnvironmentVariables();
+
+    SpringApplication.run(StandaloneBroker.class, args);
+  }
+
+  @Override
+  public void run(final String... args) throws Exception {
     final Broker broker;
 
-    if (args.length == 1) {
-      broker = createBrokerFromConfiguration(args);
+    if (EnvironmentHelper.isProductionEnvironment(springEnvironment)) {
+      // creates a broker in the base directory that will run with the configuration provided by
```

Comment:
Do you feel the method names do not reflect properly what they do?

Reply:
Yes. There are two aspects of the two methods: 
* base dir vs temp dir
* given config vs. default config

I had several rounds of renaming the methods and still got confused. That is the best I could come up with. 

I am very open for naming suggestions.

Reply:
Sorry, that was a misunderstanding completely on me :sweat_smile: I meant it as questioning if the comment , e.g. it is there because the name of the method does not properly reflect what the method does?

I would personally remove it but wanted to hear what you thought

Reply:
done